ts2mls/js/src/main/scala/ts2mls/types/Converter.scala

```diff
@@ -115,7 +115,7 @@ object Converter {
   private def convertRecord(members: Map[String, TSMemberType], typeVars: List[TSTypeParameter], parents: List[TSType],
     statics: Map[String, TSMemberType], constructorList: List[TSType], anonymous: Boolean)(implicit indent: String) = {
     val allRecs = members.toList.map((m) => m._2.modifier match {
-      case Public =>
+      case Public if !m._1.contains("#") =>
```

Comment:
Is this meant to detect private fields? If so, it should use `startsWith` not `contains`.

Reply:
So far we are reading the member's symbol name, which may be renamed by the `tsc`(e.g., `#bar` -> `__#1@#bar`). I will move the detection logic to the parsing part to get the untouched variable name rather than check during translation.

Reply:
No. This is just because in `tsc`, `private field: type` and `#field: type` have different structures. The first one uses a private keyword and an identifier, while the second one uses a private identifier(and renames the symbol).  Here is [an example](https://ts-ast-viewer.com/#code/MYGwhgzhAEBiD29oG8BQ1oGIBGYBOAXNAHYCuAttgKZ7rQAOeAlgG5gAuV0uAXkRO2bEA5qgC+qIA).
So far the parsing logic can only handle the identifiers, and that's why:
>  But so far we are reading the member's symbol name, which may be renamed by the tsc(e.g., #bar -> __#1@#bar). I will move the detection logic to the parsing part to get the untouched variable name rather than check during translation.

Reply:
All I can see from your link is that there's an `escapedName` field with containing `"__#1@#bar"`. This is obviously not something we should care about. By the way, my expectation is that this escaped name probably _is_ for the purpose of desugaring the private field into a non-private one somehow.

Note: weirdly, the TS palyground desugars your `Foo` example:
```ts
class Foo {
  #bar: number
  private baz: string
}
```
to:
```js
class Foo {
    constructor() {
        _Foo_bar.set(this, void 0);
    }
}
_Foo_bar = new WeakMap();
```

Reply:
Replaced with correct private identifiers.